azuredevops/internal/service/permissions/resource_project_permissions.go

```diff
@@ -0,0 +1,134 @@
+package permissions
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/client"
+	securityhelper "github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/utils/debug"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/utils/validate"
+)
+
+// ResourceProjectPermissions schema and implementation for project permission resource
+func ResourceProjectPermissions() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProjectPermissionsCreate,
+		Read:   resourceProjectPermissionsRead,
+		Update: resourceProjectPermissionsUpdate,
+		Delete: resourceProjectPermissionsDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceProjectPermissionsImporter,
+		},
+		Schema: securityhelper.CreatePermissionResourceSchema(map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validate.UUID,
+				Required:     true,
+				ForceNew:     true,
+			},
+		}),
+	}
+}
+
+func createProjectToken(d *schema.ResourceData) (*string, error) {
+	projectID, ok := d.GetOk("project_id")
+	if !ok {
+		return nil, fmt.Errorf("Failed to get 'project_id' from schema")
+	}
+	aclToken := fmt.Sprintf("$PROJECT:vstfs:///Classification/TeamProject/%s", projectID.(string))
+	return &aclToken, nil
+}
+
+func resourceProjectPermissionsCreate(d *schema.ResourceData, m interface{}) error {
+	debug.Wait()
+
```

Comment:
nit: Remove a empty line

Reply:
Not sure which line 😞

Reply:
Aah, I thought there was two empty lines! Nevermind